Ethernet and IP addresses pattern tests

Rangababu-R opened the pull request:
Added Test Cases for Ethernet addresses
and IP address pattern validation.
covers: Issues #104  and #87

ashutshkumr commented:
@Rangababu-R can you please check if python2 failur is legimitate on your setup.

ashutshkumr commented on tests/ethernet/eth.py:
```diff
@@ -0,0 +1,23 @@
+import utils
```
Please move all common functions to utils itself. If you think it belongs to a specific separate module, feel free to create eth.py inside `utils/`

ashutshkumr commented on tests/ethernet/eth.py:
```diff
@@ -0,0 +1,23 @@
+import utils
+
+
+def captures_ok(api, cfg, size, src, dst):
```
after moving it to utils, see if we can call it like so:
```python
utils.eth.captures_ok()
```

It's not impossible, but if it's too much effort for now, consider renaming it to add prefix.

ashutshkumr commented on tests/ethernet/eth.py:
```diff
@@ -0,0 +1,23 @@
+    for k in cap_dict:
+        for b in cap_dict[k]:
+            dst_list.append(
+                ":".join(['{:02X}'.format(byt) for byt in b[0:6]])
```
wouldn't it make sense to move it to `mac_addr_from_bytes(b[0:6])` ? If we haven something similar for IPv4 as well, we should consider the same for it as well.

Rangababu-R replied:
As this was added at protocol level, did not feel to add as definition.
will move to definition

ashutshkumr commented on tests/ethernet/test_counter_mac_addr.py:
```diff
@@ -0,0 +1,58 @@
+from abstract_open_traffic_generator import flow
+from abstract_open_traffic_generator.flow import Counter
```
let's be consistent here by simply using `flow.Counter` like all other flow components.

ashutshkumr commented on tests/ipv4/ipv4.py:
```diff
@@ -0,0 +1,18 @@
+import utils
```
Requesting approach similar to discussed in mac addr section.

ashutshkumr reviewed (changes requested):
As I didn't get a chance to review yesterday the `generate*` functions. Here's what I suggest:
- All function name should follow consistent naming based on what they return and what pattern the accept. Something like `get_mac_addrs_from_counter_pattern()` or `mac_addrs_from_counter_pattern()`. I see some names which mixed counter with list (not intentional though).
- Reduce sample size of examples for `generate*` functions from 10 to 5 or 3. It's sufficient to showcase intent.
- Add functions `mac_addr_to_num()` and `num_to_mac_addr()`. Similarly for IPv4 addr. The use of `ipaddr` library isn't quite justified IMHO. The `generate` functions should be using those functions, resulting in a more reasonable code-block.
- We refactored `stats_ok` out of tests. Why not just refactor `wait_for` and call it `wait_for_stats_ok` ?

Rangababu-R commented:
> 
> 
> As I didn't get a chance to review yesterday the `generate*` functions. Here's what I suggest:
> 
>     * All function name should follow consistent naming based on what they return and what pattern the accept. Something like `get_mac_addrs_from_counter_pattern()` or `mac_addrs_from_counter_pattern()`. I see some names which mixed counter with list (not intentional though).
> 
>     * Reduce sample size of examples for `generate*` functions from 10 to 5 or 3. It's sufficient to showcase intent.
> 
>     * Add functions `mac_addr_to_num()` and `num_to_mac_addr()`. Similarly for IPv4 addr. The use of `ipaddr` library isn't quite justified IMHO. The `generate` functions should be using those functions, resulting in a more reasonable code-block.
> 
>     * We refactored `stats_ok` out of tests. Why not just refactor `wait_for` and call it `wait_for_stats_ok` ?

Done changes with restpy and removed ipaddr lib and made counter pattern local
